load-test-framework/src/main/java/com/google/pubsub/flic/controllers/Client.java

```diff
@@ -0,0 +1,239 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+package com.google.pubsub.flic.controllers;
+
+import com.beust.jcommander.internal.Nullable;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import com.google.protobuf.Timestamp;
+import com.google.pubsub.flic.common.LatencyDistribution;
+import com.google.pubsub.flic.common.LoadtestGrpc;
+import com.google.pubsub.flic.common.LoadtestProto;
+import com.google.pubsub.flic.common.LoadtestProto.KafkaOptions;
+import com.google.pubsub.flic.common.LoadtestProto.PubsubOptions;
+import com.google.pubsub.flic.common.LoadtestProto.StartRequest;
+import com.google.pubsub.flic.common.LoadtestProto.StartResponse;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class Client {
+  static final String topicPrefix = "cloud-pubsub-loadtest-";
+  private static final Logger log = LoggerFactory.getLogger(Client.class);
+  private static final int port = 5000;
+  public static int messageSize;
+  public static int requestRate;
+  public static Timestamp startTime;
+  public static int loadtestLengthSeconds;
+  public static int batchSize;
+  public static String broker;
+  public static int maxOutstandingRequests;
+  public static long burnInTimeMillis;
+  public static int numberOfMessages = 0;
+  private final ClientType clientType;
```

Comment:
I'm confused as to why some of these settings are static and some are not. How did you decide?